CMakeLists.txt

```
@@ -38,29 +42,31 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-# Enable all the warnings we can. We need noncall exceptions so we can throw exceptions from signal handlers. This
-# allows us to catch null pointer exceptions. no-int-in-bool-context is to silence a warning from Eigen.
-add_compile_options(-Wall -Wpedantic -Wextra -Wno-int-in-bool-context -fnon-call-exceptions)
-
-# C++17 allows eigen to not need to worry about alignment
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-
-# Make sure our binaries can find our libraries both in docker and on the robot
-list(APPEND CMAKE_INSTALL_RPATH /home/nubots/toolchain/)
-list(APPEND CMAKE_INSTALL_RPATH /usr/local/lib)
-
-# Disable armadillo bounds checking in release and min size release builds
-list(APPEND CMAKE_CXX_FLAGS_RELEASE "-DARMA_NO_DEBUG")
-list(APPEND CMAKE_CXX_FLAGS_MINSIZEREL "-DARMA_NO_DEBUG")
-
-# Default not to run the clang-tidy checks
+# Default not to run the clang-tidy checks, default to whatever our CI_BUILD is
 option(ENABLE_CLANG_TIDY "Enable building with clang-tidy checks." OFF)
-if(ENABLE_CLANG_TIDY)
-  set(CMAKE_CXX_CLANG_TIDY clang-tidy)
-  set(CMAKE_C_CLANG_TIDY clang-tidy)
-endif()
+if(ENABLE_CLANG_TIDY OR CI_BUILD)
+  find_package(PythonInterp 3 REQUIRED)
+  set(CMAKE_CXX_CLANG_TIDY "${PYTHON_EXECUTABLE}" "${PROJECT_SOURCE_DIR}/cmake/scripts/clang-tidy.py"
+                           "${PROJECT_BINARY_DIR}/clang-tidy-fixes" clang-tidy
+  )
+  set(CMAKE_C_CLANG_TIDY ${CMAKE_CXX_CLANG_TIDY})
+
+  # Create a target that will apply clang-tidy fixes to the codebase
+  add_custom_target(
+    apply-clang-tidy
+    COMMAND clang-apply-replacements --format --style=file --style-config="${PROJECT_SOURCE_DIR}"
+            --remove-change-desc-files "${PROJECT_BINARY_DIR}/clang-tidy-fixes"
+    COMMENT "Applying fixes from clang-tidy to the codebase."
+  )
+endif(ENABLE_CLANG_TIDY OR CI_BUILD)
```

We don't use endif conditions any more
```suggestion
endif()
```